components/service/glean/src/main/java/mozilla/components/service/glean/firstrun/FirstRunDetector.kt

```diff
@@ -0,0 +1,26 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.firstrun
+
+/**
+ * Defines the interface for a service detecting if this
+ * is the first time the glean is used by the app.
+ */
+internal interface FirstRunDetector {
+    /**
+     * Detects whether or not this is the first time
+     * Glean is used in this application
+     *
+     * @return true if this is the first run, false otherwise.
+     */
+    fun isFirstRun(): Boolean
+
+    /**
+     * Clears up the "next run" state. The next
+     * time [isFirstRun] is called, it will report
+     * a first run.
+     */
+    fun reset()
```

Comment:
The `reset` method confused me: Is this about the first run after an app isntallation or first run in a process or a very component specific meaning of first run?

Reply:
Right! So, say the application wants to "remove the user profile" or "user data", or opt-out of `glean`. This will make sure, next time glean is opted in again, we go through any "first time" configuration again. In our specific case, this would internally trigger generating a new `client id`.